hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/HoodieWriteHelper.java

```diff
@@ -78,7 +79,8 @@ protected HoodieData<HoodieRecord<T>> doDeduplicateRecords(
         throw new HoodieException(String.format("Error to merge two records, %s, %s", rec1, rec2), e);
       }
       HoodieKey reducedKey = rec1.getData().equals(reducedRecord.getData()) ? rec1.getKey() : rec2.getKey();
-      return reducedRecord.newInstance(reducedKey);
+      HoodieOperation operation = rec1.getData().equals(reducedRecord.getData()) ? rec1.getOperation() : rec2.getOperation();
```

Comment:
Why we compare the data between rec1 and rec2  here instead of comparing the reference between rec1 and reducedRecord just as what  FlinkWriteHelper do?

Reply:
This is some risk here because not all payload merger confirm the rules, for e.g. the `PartialUpdateAvroPayload` actually returns a new object each time, the tricky part is the operation does not affect because all the operations should be upsert for partial update scenarios.